packages/lesswrong/server/notificationCallbacks.tsx

```diff
@@ -458,25 +458,31 @@ getCollectionHooks("Posts").newAsync.add(async function PostsNewNotifyUsersShare
   await createNotifications({userIds, notificationType: "postSharedWithUser", documentType: "post", documentId: _id})
 });
 
-getCollectionHooks("Posts").newAsync.add(async function CoauthorRequestNotifications(post: DbPost) {
-  const { _id, coauthorStatuses = [], hasCoauthorPermission } = post;
+getCollectionHooks("Posts").newAsync.add(async function CreateCoauthorRequestNotifications(post: DbPost) {
```

Comment:
Should these be newAsync / updateAsync or newAfter / updateAfter ?

It doesn't really matter, but I'd nitpickingly make them consistent.

See packages/lesswrong/server/vulcan-lib/mutators.ts for the behavior.

Note that I _believe_ that the async commands are all run synchronously (not, like, blocking the main thread, but all one after the other and all before the mutation will return). This is because @jimrandomh made the call that it wasn't worth the race conditions to do them properly async.

Reply:
That's a fair point. I just checked and you're totally correct that the async callbacks are currently synchronous. I changed the code to use `newAfter` and `updateAfter` just because we mutate the post itself in the callback which could be a bit weird if we ever change the async callbacks to actually be async.